Source/FaustProcessor.cpp

```diff
@@ -41,7 +41,9 @@ FaustProcessor::FaustProcessor(std::string newUniqueName, double sampleRate, int
 
 FaustProcessor::~FaustProcessor() {
 	clear();
-	deleteAllDSPFactories();
+	if (myMidiIterator) {
```

Comment:
If `myMidiIterator` is properly initialized with `nullptr`, you don't need to test it, because `delete nullptr` is allowed.

Reply:
Thanks I'll try this in the next round of CI tests